artemis-commons/src/main/java/org/apache/activemq/artemis/utils/collections/TypedProperties.java

```diff
@@ -318,6 +320,33 @@ public synchronized boolean containsProperty(final SimpleString key) {
       }
    }
 
+   public synchronized boolean cleanupInternalProperties(Predicate<SimpleString> propertyNamePredicate) {
+      if (!internalProperties) {
```

Comment:
It may have them from legacy someone put it there, but if you were designing a collections class, you'd design it in a fashion so it focussed just on the logic it needs to have so you ensure its good. 

If anything that field, the flag and the method check hasInternalProperties really should move upto CoreMessage, as its only used there.

Reply:
I agree: in that case I would move the other `Message`ish things outside `TypedProperties`.
I was hoping for a less impactfull change TBH, but you really got a point about this. Let me check if I can perform a surgical removal to make the code simpler and cleaner :+1:

Reply:
@michaelandrepearce I'm working on `CoreMessage` to inject into it both `internalProperties` and the cleanupLogic, but TBH the concurrent behaviour is not trivial: it has to be consistent with `TypedProperties` content assuming that the volatile reference of it into `CoreMessage` could change: I'm not sure it would be that easy to change it

Reply:
I have pushed a new version of the change that's using a private subclass of 'TypedProperties' where needed to allow atomic updates of the `internalProperties` flag: I'm not very about about the factory method on 'CoreMessage' to allow the client messages to not use that subclass: this is an additional optimisation given that the original code where always using 'TypedProperties' that where chacking on each added property if it was an 'internal" one.

Reply:
I have added a super tiny optimization on `PacketImpl` turning a method into a static one: using https://github.com/AdoptOpenJDK/jitwatch to read the compilation logs seems to help the inlining of the method (for free). I would prefer to maintain separated the optimizations, but a single PR for a single word change (not a bug fix) seems really too much for me...wdyt?

Reply:
tbh id rather we do one refactor. so +1 for me.